lib/async.js

```diff
@@ -5,7 +5,19 @@ var nodeModulesPaths = require('./node-modules-paths.js');
 var normalizeOptions = require('./normalize-options.js');
 var isCore = require('./is-core');
 
-var realpathFS = fs.realpath && typeof fs.realpath.native === 'function' ? fs.realpath.native : fs.realpath;
+function realpathFS(path, callback) {
+    if (typeof fs.realpath.native === 'function') {
+        fs.realpath.native(path, function (realpathErr, realPath) {
+            if (realpathErr && realpathErr.errno === -4068) {
+                fs.realpath(path, callback);
+            } else {
+                return callback(realpathErr, realPath);
+            }
+        });
```

Comment:
this means we pay a performance cost every time. is this necessary, or on a given system where it fails, will it always fail?

Reply:
This code is designated for the `postcss-import` on RAM Disk. This means the I/O speed is significantly improved. The production build of my Angular project takes 60-120 secs with `fs.realpath.native()` on my 2TB Seagate FireCuda SSHD but it only takes ~60 secs on RAM Disk mounted by imDisk with `fs.realpath()` (*where `fs.realpath.native()` is illegal operation*). So in general, no performance costs are paid.

Reply:
I'm not sure how that's possible - this is always calling realpath.native, and when it errors, is *adding* the cost of calling realpath.

(to be clear, I have no idea what imDisk is, and the last time I heard anything about RAM Disks was in the late 1990s)

Reply:
imDisk is a software to create a virtual disk using RAM on Windows: https://imdisk.en.lo4d.com/windows 

The cost of calling is the trade-off for I/O speed and it only adding the call on illegal operation error.

You can clone this repository using Windows on both RAM Disk created by imDisk and your HDD/SSHD/SSD https://github.com/scttcper/ngx-toastr. After the clone, run ```npm i && npm start```. On RAM Disk, the server will fail and you will get 
![image](https://user-images.githubusercontent.com/14126301/94336429-dddc1c80-ffa8-11ea-8be8-a184cedfa395.png)

The cause is from `postcss-import` using this repository to resolve the imported files from *node_modules* in *src/style.scss*. So if you alter *./node_modules/resolve/lib/async.js*, the error is gone.

Reply:
> imDisk is a software to create a virtual disk using RAM on Windows: https://imdisk.en.lo4d.com/windows
> 
> The cost of calling is the trade-off for I/O speed and it only adding the call on illegal operation error.
> 
> You can clone this repository using Windows on both RAM Disk created by imDisk and your HDD/SSHD/SSD https://github.com/scttcper/ngx-toastr. After the clone, run `npm i && npm start`. On RAM Disk, the server will fail and you will get
> ![image](https://user-images.githubusercontent.com/14126301/94336429-dddc1c80-ffa8-11ea-8be8-a184cedfa395.png)
> 
> The cause is from `postcss-import` using this repository to resolve the imported files from _node_modules_ in _src/style.scss_. So if you alter _./node_modules/resolve/lib/async.js_, the error is gone.

It's ```npm i ; npm start``` on PowerShell.

Reply:
I'm concerned about:
1. adding a fix without a regression test, because there's no protection from breaking it again
1. adding a complicated fix that might slow down the majority case to service a *very* niche edge case